server/src/ai/ai.functions.ts

```diff
@@ -22,27 +22,104 @@ const openai = new OpenAI({
+          You are given a collection of course reviews provided where each review is separated by a /. You
+          will then complete two tasks. First you should generate a 50 word summary of all reviews. Then 
+          you should create 5 adjectives describing the lectures, assignments, professor, skills, and resources,
+          along with their connotations (positive, negative, neutral). Please only pick one adjective for each.
+          
+          Please provide the summary and tags in the following format:
+          Summary: [50-word summary]
+          Tags:
+          'Lectures: [adjective] (positive/negative/neutral),
+          Assignments: [adjective] (positive/negative/neutral),
+          Professor: [adjective] (positive/negative/neutral),
+          Skills: [adjective] (positive/negative/neutral),
+          Resources: [adjective] (positive/negative/neutral)'.
+        `
+      },
       { role: "user", content: text }
     ],
   });
-  return completion.choices[0].message.content;
+
+  const response = completion.choices[0].message.content;
+  const summaryMatch = response.match(/Summary: ([\s\S]*?)(?=Tags)/);
+  const summary = summaryMatch ? summaryMatch[1].trim() : "Summary not found.";
+  const tagsMatch = response.match(/Tags:\s*([\s\S]*)/);
+  const tags = tagsMatch ? tagsMatch[1] : "";
+  const tagsObject: { [key: string]: [string, string] } = {};
+  tags.split(',').forEach(item => {
+    const match = item.match(/(\w+): (.+) \((.+)\)/);
+    if (match) {
+      const category = match[1].trim();
+      const adjective = match[2].trim();
+      const connotation = match[3].trim();
+      if (adjective !== "N/A" && connotation != "N/A") {
+        tagsObject[category] = [adjective, connotation];
+      }
+    } else {
+      console.error("Unexpected format: ", item);
+    }
+  });
+
+  return {
+    summary: summary,
+    tags: tagsObject
+  };
 }
 
+/**
+ * updateCourseWithAI.
+ * 
+ * Takes in a courseId and uses that ID to get all reviews from a course to then
+ * generate a summary and 5 tags for those reviews. Then updates the classSummary
+ * and classTags fields in the database for the corresponding course with the 
+ * newly generated summary and tags. Also resets the freshness count for the course
+ * back to 0. Returns a boolean indicating the success of this update.
+ * @params a courseId in the form of a string
+ * @returns true if update was successful and false if something went wrong
+ * 
+ */
+const updateCourseWithAI = async (courseId: string) => {
+  try {
+    const courseReviews = await Reviews.find({ class: courseId }).exec();
+
+    const reviewsText = courseReviews.map(review => review.text || "").join(' / ');
+
+    const { summary, tags } = await summarize(reviewsText);
+
+    await Classes.updateOne(
+      { _id: courseId },
+      { $set: { classSummary: summary, classTags: tags, freshness: 0 } }
+    );
+
+    console.log(`Course ${courseId} updated successfully with summary and tags.`);
```

Comment:
This would definitely be helpful for readability purposes. I think something like

`const course = await Classes.findOne({ _id: courseId })`
`const courseName = course.classSub.toUppercase() + ' ' + course.classNum;`

could work, and then put that variable in the print statement